### PR TITLE
 Fix conditional expressions in all places

### LINKED ISSUE #24294: Fix conditional expressions in all places
When there is `expr1 ? ident : expr2`, parser needs to termination without treating `ident:` as a qualified identifier.

Currently this scenario works. However, this middle-expr can be many other things as well such as `check ident`, `<T> ident`, etc. So we need to fix this for any expression that can ends with a n identifier.

Some of the issues were fixed in https://github.com/ballerina-platform/ballerina-lang/pull/24279/commits/63c5f08e8bfb1264361cf96f78769034492de40d. But need to find and fix remaining ones.

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #24294 

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by lochana-chathura** [comment]
Most of the changes are from last commit (https://github.com/ballerina-platform/ballerina-lang/pull/24407/commits/a76b8aa955d8f4a5a188a7aded72b5eec3ef5b05) as type-descriptor is referred in many places.

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -2319,7 +2327,7 @@ private STNode parseTypeDescriptorInternal(SyntaxKind tokenKind, ParserRuleConte
                     return solution.recoveredNode;
                 }
 
-                return parseTypeDescriptorInternal(solution.tokenKind, context);
+                return parseTypeDescriptorInternal(solution.tokenKind, context, false);
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:2330]: Shouldn't we pass `isInConditionalExpr` as the last arg?

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java]: yh It should be. thanks. Fixed with bbc37f6. 
btw just found out that parser crashes for
```
function main() {
    int x = a?b:c
}
```
missing semicolon in conditional-expr. I'll have a look

Comment (lochana-chathura) [reply compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java]: Latter issue fixed with aecf63b

**Pull request merged**